src/coreclr/src/ToolBox/superpmi/superpmi-shared/compileresult.h

```diff
@@ -11,6 +11,47 @@
 #include "runtimedetails.h"
 #include "lightweightmap.h"
 
+// MemoryTracker: a very simple allocator and tracker of allocated memory, so it can be deleted when needed.
+class MemoryTracker
+{
+public:
+    MemoryTracker() : m_pHead(nullptr) {}
+    ~MemoryTracker() { freeAll(); }
+
+    void* allocate(size_t sizeInBytes)
+    {
+        BYTE* pNew = new BYTE[sizeInBytes];
+        m_pHead = new MemoryNode(pNew, m_pHead);    // Prepend this new one to the tracked memory list.
```

Comment:
Nit: This has memory leak if `new MemoryNode` throws, but I assume that we do not care about hardening like this in superpmi.

Reply:
Yes, I'm not too worried about hardening superpmi for OOM